cmd/dex-worker/main.go

```diff
@@ -48,6 +48,7 @@ func main() {
 
 	enableRegistration := fs.Bool("enable-registration", false, "Allows users to self-register")
 	enableClientRegistration := fs.Bool("enable-client-registration", false, "Allow dynamic registration of clients")
+	registerOnFirstLogin := fs.Bool("register-on-first-login", false, "When a user logs in through a federated identity service, automatically register them if they don't have an account. This flag is independent of enable-registration.")
```

Comment:
That's a lot of what @chancez was getting at. We're trying to express three types of registration with two flags.

* Require registration
* Require local connector and invites
* Require upstream registration